fix(docs): contribution documentation cleanup (#40761)
airbytehq/airbyte: docs/contributing-to-airbyte/README.md

- New API sources built with the low-code CDK or Connector Builder.
- Migrating existing sources from Python to the low-code CDK.
- Bug fixes, features, and enhancements to the following database sources: Postgres, MySQL, MSSQL.
- Bug fixes to the following destinations: BigQuery, Snowflake, Redshift, S3, and Postgres.
- Helm Charts features, bug fixes, and other platform bug fixes.

:::warning
Airbyte is undergoing a major revamp of the shared core Java destinations codebase, with plans to release a new CDK in 2024.
We are actively working on improving usability, speed (through asynchronous loading), and implementing [Typing and Deduplication](/using-airbyte/core-concepts/typing-deduping) (Destinations V2).
We're not actively reviewing/accepting new Java connectors for now.
:::

#### The usual workflow of code contribution is:

1. Fork the Airbyte repository.
2. Clone the repository locally.
3. Create a branch for your feature/bug fix with the format `{YOUR_USERNAME}/{FEATURE/BUG}` (e.g. `jdoe/source-stock-api-stream-fix`)
4. Make and commit changes.
5. Push your local branch to your fork.
6. Submit a Pull Request so that we can review your changes.

> 7. [Link an existing Issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) that does not include the `needs triage` label to your Pull Request. Pull requests without an issue attached to take longer to review.

Reviewer: ```suggestion
7. [Link an existing Issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) that does not include the `needs triage` label to your Pull Request. Pull requests without an issue attached take longer to review.
```

8. Write a PR title and description that follows the [Pull Request Handbook](./resources/pull-requests-handbook.md).
9. An Airbyte maintainer will trigger the CI tests for you and review the code.
10. Review and respond to feedback and questions by Airbyte maintainers.
11. Merge the contribution.

Guidelines to common code contributions:

- [Contribute a code change to an existing connector](change-cdk-connector.md)
- [Contribute a new connector](submit-new-connector.md)

## Documentation

> We welcome Pull Requests that enhance the grammar, structure, or fix typos in our documentation! Check the [Updating Documentation](writing-docs.md) guide for submitting documentation changes.

Reviewer: ```suggestion
We welcome Pull Requests that fix typos or enhance the grammar and structure of our documentation! Check the [Updating Documentation](writing-docs.md) guide for details on submitting documentation changes.
```

> If you're working on connector documentation, make sure you cover:

Reviewer: Missing a list of what to cover on connectors?

> If you're working on connector documentation, make sure you cover:

Author: ```suggestion
```

The following highlights from the [Google developer documentation style guide](https://developers.google.com/style) are helpful for new writers:

- [Be conversational and friendly without being frivolous](https://developers.google.com/style/tone)
- [Use second person](https://developers.google.com/style/person)
- [Use active voice](https://developers.google.com/style/voice)
- [Put conditional clauses before instructions](https://developers.google.com/style/clause-order)
- [Put UI elements in bold](https://developers.google.com/style/ui-elements)
- [Write inclusive documentation](https://developers.google.com/style/inclusive-documentation)
- [Don't pre-announce anything in documentation](https://developers.google.com/style/future)

Guideline for visuals: Use links to videos instead of screenshots (Reason: Users are more forgiving of outdated videos than screenshots).

> For examples of good connector docs, see the [Salesforce source connector doc](https://docs.airbyte.com/integrations/sources/salesforce) and [Snowflake destination connector doc](https://docs.airbyte.com/integrations/destinations/snowflake)

Reviewer: I was thinking that it would be useful to point out exemplary connector docs, so this is a nice addition!

## Community Content

> We welcome contributions as new tutorials / showcases / articles, or to any of the existing guides on our tutorials page. Head to this repo dedicated to community content: [Write for the Community](https://github.com/airbytehq/write-for-the-community).

Reviewer: ```suggestion
We welcome contributions as new tutorials / showcases / articles, or as enhancements to any of the existing guides on our tutorials page. Head to this repo dedicated to community content: [Write for the Community](https://github.com/airbytehq/write-for-the-community).
```

Feel free to submit a pull request in this repo, if you have something to add even if it's not related to anything mentioned above.

## Engage with the Community

> Another crucial way to contribute is by reporting bugs and helping other users in the community. You're welcome to join the [Community Slack](https://slack.airbyte.io). Refer to [Issues and Feature Requests](issues-and-requests.md) guide to learn about best ways to report bugs.

Reviewer: ```suggestion
Another crucial way to contribute is by reporting bugs and helping other users in the community. You're welcome to join the [Community Slack](https://slack.airbyte.io). Refer to our [Issues and Feature Requests](issues-and-requests.md) guide to learn about the best ways to report bugs.
```
